src/api.c

```diff
@@ -102,6 +102,10 @@ static unsigned int _seccomp_api_update(void)
 	    sys_chk_seccomp_flag(SECCOMP_FILTER_FLAG_TSYNC) == 1)
 		level = 2;
 
+	/* level 3 */
+	if (sys_chk_seccomp_flag(SECCOMP_FILTER_FLAG_LOG) == 1)
```

Comment:
The API level is a "everything and below" indicator, meaning that if we say we are operating at level 3 it means we support *everything* at level 3 and below.  For that reason I think this check should be something like the following:

    if ((level == 2) && (sys_chk_seccomp_flag(SECCOMP_FILTER_FLAG_LOG) == 1))

Granted, I think in most cases if FLAG_LOG is supported then everything else will be supported as well, but I would rather be safe.

Reply:
I agree. Nice catch!